### DISCUSSION TITLE
"Автоконец" у ищущих

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g2nr_unofficial_update#374] [Ideas]
Не помню обсуждали мы или нет, но я ищущим убирал "конец"  с ним они стали бодрее и веселее
https://www.youtube.com/watch?v=htoIvEu87Z8

правка в коде там была простая с выходом через **AI_StartState**

```D
func void DIA_BridgeDementor_Info()
{
	Wld_PlayEffect("DEMENTOR_FX",hero,hero,0,0,0,FALSE);
	Wld_PlayEffect("spellFX_Fear",self,self,0,0,0,FALSE);
	AI_PlayAni(self,"T_PRACTICEMAGIC5");
	AI_Output(self,other,"DIA_BridgeDementor_19_00");	//Ìû çíàëè, ÷òî òû ïðèäåøü!
	AI_Output(self,other,"DIA_BridgeDementor_19_01");	//Ìîé Õîçÿèí îäåðæèò ïîáåäó è ñêëîíèò âåñü ìèð ê ñâîèì íîãàì.
	AI_Output(self,other,"DIA_BridgeDementor_19_02");	//Òû íàèâíûé èäèîò! Òû ïîíÿòèÿ íå èìååøü, íà ÷òî òû çàìàõíóëñÿ, è ÿ íàêàæó òåáÿ çà ýòî çäåñü è ñåé÷àñ!
	Npc_SetRefuseTalk(self,30);
	AI_StartState	(self, DIA_BridgeDementor_EXIT_Info,1,"");
};
```

- https://github.com/Kor-Angar/G2_NotR_Kor_Angar/commit/d439c44173875b82cef857c089572d525cae2fb7 
- https://github.com/Kor-Angar/G2_NotR_Kor_Angar/commit/776537c26ef7b27966feb0bb8c6513c72e4490b1

сейчас играл и прям такой расслабленный был :-) ибо можно подождать :-) ты ждёшь, Ищущие ждут, все ждут :-)

**Comment by Kor-Angar** [comment]
>  ибо можно подождать :-) ты ждёшь, Ищущие ждут, все ждут :-)

![20210906205150_1](https://user-images.githubusercontent.com/49514764/132258731-c3023e02-1a49-4582-89dc-1579a5718c76.jpg)

**Comment by Kor-Angar** [comment]
@dosinabox получится добавить в свежую сборку?

**Reply by dosinabox** [reply]
Добавлю, заодно и проверим

**Reply by Kor-Angar** [reply]
получилось добавить? :-)

**Reply by dosinabox** [reply]
да
https://github.com/dosinabox/g2nr_unofficial_update/commit/29048590f48d123a3d39940e2c97f37f8e0911ef

**Reply by dosinabox** [reply]
DementorsDialogeAutoEndDisabled - на случай проблем, чтобы можно было через марвин вернуть как было.

**Reply by Kor-Angar** [reply]
ага.. я так и понял про DementorsDialogeAutoEndDisabled

**Comment by Kor-Angar** [comment]
см. так же https://github.com/dosinabox/g2nr_unofficial_update/discussions/509

**Comment by Kor-Angar** [comment]
одержимые

```D
func void BAI_NpcClearObsessionByDMT()
{
    Hlp_PrintConsole(Str_Format("   BAI_NpcClearObsessionByDMT (%s %i)",self.name,self.id));
    B_NpcClearObsessionByDMT(self);
};


func void B_NpcObsessedByDMT (VAR C_NPC medium)
{
    Hlp_PrintConsole(Str_Format("   B_NpcObsessedByDMT (%s %i)",medium.name,medium.id));
    B_NpcObsessedByDMT_old(medium);
   
	if (NpcObsessedByDMT == TRUE)
    {
        AI_StartState(medium,BAI_NpcClearObsessionByDMT,0,""); 
        //Npc_GiveInfo(medium,0);   //
    };
};

```

**Reply by Kor-Angar** [reply]
![20230228022024_1](https://user-images.githubusercontent.com/49514764/221915105-932b3204-1b64-462f-b301-8e8568332a0d.jpg)

**Reply by dosinabox** [reply]
кого-то забыли?

**Reply by Kor-Angar** [reply]
одержимых. автозапуск "конца" для них 

```D
func void B_NpcObsessedByDMT (VAR C_NPC medium)
{
    ...
    
    ...
   
    if (NpcObsessedByDMT == TRUE)
    {
        AI_StartState(medium,BAI_NpcClearObsessionByDMT,0,""); 
    };
};
```

**Comment by Kor-Angar** [comment]
у тебя в `B_NpcClearObsessionByDMT`  почему то не удаляется фейковая броня `ITAR_Fake_Dementor`.
